src/Concordium/Client/Output.hs

```diff
@@ -699,6 +728,26 @@ showEvent verbose = \case
     showDelegationTarget Types.DelegatePassive = "Passive delegation"
     showDelegationTarget (Types.DelegateToBaker bid) = "Baker ID " ++ show bid
 
+    showLoggedEvents :: [Wasm.ContractEvent] -> String
+    showLoggedEvents [] = "No contract events were emitted "
+    showLoggedEvents evs = [i|#{length evs} contract events were emitted|]
+      <> if isNothing stM
+         then [i| but no event schema was provided nor found in the contract module. |]
+         else [i|, of which #{length $ filter isJust $ map snd loggedEvents} were succesfully parsed using schema. |]
+      <> [i|Got:\n|]
+      <> intercalate "\n" [ if isJust json
+                            then [i|Event(parsed):\n#{indentBy 4 $ fromMaybe' json}|]
+                            else [i|Event(raw): #{hex}\n|]
+                            | (hex, json) <- loggedEvents ]
+        where
+          fromMaybe' :: Maybe AE.Value -> String
+          fromMaybe' = \case Nothing -> "Unable to decode."; Just j -> showSortedPrettyJSON j
```

Comment:
Please do not introduce new syntax. We do not use semicolons in Haskell to squeeze things into a single line.

Use the prevailing code style. The simpler the better.

Reply:
This is also a very strange function as is. And if it remains needs to have rationale in its documentation.

Reply:
I agree. This function as a whole is terrible, and I'll rewrite it.